test/distributed/fsdp/test_fsdp_state_dict.py

```diff
@@ -513,6 +555,69 @@ def test_basic_save_and_load_state_dict(
 
             self._compare_models(model, model_new, self.assertEqual, check_fp16=fp16)
 
+    @skip_if_lt_x_gpu(2)
+    @parametrize("state_dict_type", _SUPPORTED_STATE_DICT_IMPLS)
+    @parametrize(
+        "cpu_offload",
+        [CPUOffload(offload_params=True), CPUOffload(offload_params=False)],
+    )
+    @parametrize("mixed_precision", [True, False])
+    @parametrize("state_dict_rank0_and_offload", [True, False])
+    def test_buffers_save_and_load_state_dict(
+        self,
+        state_dict_type: StateDictType,
+        cpu_offload: bool,
+        mixed_precision: bool,
+        state_dict_rank0_and_offload: bool,
+    ):
+        """
+        Tests that we can save a state_dict and load it for modules with persistent buffers, including
+        in the context of non-default mixed precision, different ``state_dict_type`` s and CPU offloading.
+        """
+        if state_dict_rank0_and_offload and state_dict_type != "state_dict":
+            return  # not supported
+        mixed_precision = (
+            MixedPrecision(
+                param_dtype=torch.float16,
+                reduce_dtype=torch.float16,
+                buffer_dtype=torch.float16,
+            )
+            if mixed_precision
+            else None
+        )
+        model_call = partial(
```

Comment:
what's the need for partial if we're directly using this result after?

Reply:
I am guessing this partial application is being saved for the additional `model_new = model_call()` below.

Reply:
Yep, I was using it in model_new but can switch to avoid using partial pass the the model function directly if you guys prefer that.